### PR TITLE
refactor interface

### PR DESCRIPTION
**Pull request by andiwand**

### CODE REVIEW

File: odr/include/odr/Reader.h [hunk 1]
```diff
@@ -33,7 +34,10 @@ class OpenDocumentReader final {
   bool canEdit() const noexcept;
   bool canSave() const noexcept;
   bool canSave(bool encrypted) const noexcept;
-  const FileMeta &getMeta() const noexcept;
+
+  bool isEncrypted() const noexcept;
```
Comment (TomTasche) [inline odr/include/odr/Reader.h:38]: should this be called encrypted to be consistent with your other naming?

File: odr/include/odr/Reader.h [hunk 2]
```diff
@@ -10,15 +10,16 @@ enum class FileType;
...
 public:
-  static std::string getVersion() noexcept;
-  static std::string getCommit() noexcept;
+  static std::string version() noexcept;
+  static std::string commit() noexcept;
 
-  OpenDocumentReader();
-  ~OpenDocumentReader();
+  static FileType guessType(const std::string &path) noexcept;
+  static FileMeta fetchMeta(const std::string &path) noexcept;
```
Comment (TomTasche) [inline odr/include/odr/Reader.h:19]: parseMeta?

Comment (TomTasche) [reply odr/include/odr/Reader.h:19]: and parseType?

File: odr/include/odr/Reader.h (hunk 1 above)
Comment (TomTasche) [reply odr/include/odr/Reader.h:38]: same for all other is*-methods

Comment (andiwand) [reply odr/include/odr/Reader.h:38]: you might be right but i am not sure about the is prefix :thinking: i will think about it

File: odr/include/odr/Reader.h (hunk 2 above)
Comment (andiwand) [reply odr/include/odr/Reader.h:19]: maybe `readMeta` `readType` ? i dont like `parse*` because for me it sounds like it is handling text files

Comment (TomTasche) [reply odr/include/odr/Reader.h:19]: yeah read is okay

**Pull request merged**